server/channels/store/sqlstore/file_info_store.go

```diff
@@ -455,6 +455,27 @@ func (fs SqlFileInfoStore) DeleteForPost(rctx request.CTX, postId string) (strin
 	return postId, nil
 }
 
+func (fs SqlFileInfoStore) DeleteForPostByIds(rctx request.CTX, postId string, fileIDs []string) error {
+	query := fs.getQueryBuilder().
+		Update("FileInfo").
+		Set("DeleteAt", model.GetMillis()).
+		Where(sq.Eq{
+			"PostId": postId,
+			"Id":     fileIDs,
```

Comment:
Yeah its possible for multiple post IDs to point to same file, but a file would always point to a single post. Not sure where we use this check though. When attaching a file to a post, we verify that the file is not attached to any post so that keeps it safe.